File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.java
class Solution {
    public int solution(int num) {
        
        int count = 0;
        
        long copyNum = (long) num;
        
        while(copyNum > 1) {
            
            copyNum = copyNum % 2 == 0 ? copyNum / 2 : copyNum * 3 + 1;
            count++;
            
            if (count > 500) {
                break;
            }
        }
        
        return count == 0 ? 0 : count > 500 ? -1 : count;
    }
}

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.kt
class Solution {
	fun solution(num: Int): Int {
		var answer = 0

		var col = num.toLong()

		while (answer <= 501) {
			if (col == 1L) {
				break
			}
			
			if (col % 2L == 0L) {
				col /= 2L
				answer++
			} else {
				col = col * 3 + 1
				answer++
			}
		}

		return if (answer > 500) -1 else answer
	}
}